pkg/providers/vmware/client/client_test.go

```diff
@@ -0,0 +1,81 @@
+package client_test
+
+import (
+	"github.com/kubevirt/vm-import-operator/pkg/providers/vmware/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/simulator"
+)
+
+var _ = Describe("Test VMware rich client", func() {
```

Comment:
Please think how to mock vmware for functional tests. We created fakeovirt and I think you need something similar.

Reply:
I believe the simulator package in the vmware client library can handle that. I'll continue to investigate it.